### ISSUE TITLE
I would like to know whether the provided segmentation model weights were trained based on the MIO dataset, or the MIO++ dataset, or a combination of both?

### ISSUE THREAD

**Issue by zjtde1990** [PROPHETE-pro/MaterialSeg3D#2]
I would like to know whether the provided segmentation model weights were trained based on the MIO dataset, or the MIO++ dataset, or a combination of both?

**Comment by PROPHETE-pro** [comment]
Thank you for the interest!
Currently, the announced segmentation weights are trained on MIO datasets. We only public the MIO++ dataset with the raw data distributions so far, and the weights will be public in the future, but it will take times.

**Comment by zjtde1990** [comment]
Thank you for your response. However, I still have some doubts. When I used the provided model weights and the test set of the MIO dataset to test the segmentation results, I found that the actual test performance indicators were much higher than those in the paper. Could this be correct? The category I tested was "building".

![Image](https://github.com/user-attachments/assets/8cb7e01e-96f7-4594-af11-a228a085ec93)

![Image](https://github.com/user-attachments/assets/b62f59ff-c54d-451c-905a-afeaa4e889df)

**Comment by PROPHETE-pro** [comment]
Thanks for your feedback, if possible, can you try to evaluate the performance on the validation set and provide another results? We have made some minor adjustments before the final version of the dataset, if there are indeed some problems, thanks for helping us to find out and I will be working it out

**Comment by zjtde1990** [comment]
I attempted to retrain using the provided code and the MIO dataset, but I encountered very serious problems. I attempted to retrain using the buildings data and the corresponding configuration files, but I found that my training results were very low compared to the provided model metrics. Could you offer some ideas?

![Image](https://github.com/user-attachments/assets/d026cf6e-b461-473b-b3f8-649d56b4ee27)

**Comment by PROPHETE-pro** [comment]
Thank you for pointing out the problem. Our dataset was initialled labelled with 20 classes and further refined to 16 (14 + bg + other) classes. It seems that we have uploaded the early version of the ground truth label and the config. Here I can provide you a mapping code that you can process the GT by yourself, and the correct config files.

`import numpy as np
import pdb
import cv2
import os
import json

def getFileList(dir, Filelist, ext=None, skip=None, spec=None):
    newDir = dir
    if os.path.isfile(dir):
        if ext is None:
            Filelist.append(dir)
        else:
            if ext in dir[-3:]:
                Filelist.append(dir)

    elif os.path.isdir(dir):
        for s in os.listdir(dir):
            if os.path.isdir(os.path.join(dir, s)):
                newDir = os.path.join(dir, s)
                getFileList(newDir, Filelist, ext, skip, spec)

            else:
                acpt = True
                if skip is not None:
                    for skipi in skip:
                        if skipi in s:
                            acpt = False
                            break

                if acpt == False:
                    continue
                else:
                    sp = False
                    if spec is not None:
                        for speci in spec:
                            if speci in s:
                                sp = True
                                break

                    else:
                        sp = True

                    if sp == False:
                        continue
                    else:
                        newDir = os.path.join(dir, s)
                        getFileList(newDir, Filelist, ext, skip, spec)

    return Filelist


def transfer2(image, mapping):
    image_base = image.copy()

    for i in mapping.keys():
        image[image_base == i] = mapping[i]

    return image


image_dir = "/path-to-gt/"
transfer_image_written_dir = "/path-to-mapping-gt/"

mapping = {3:2, 4:2, 5:3, 6:4, 7:5, 8:6, 9:6, 10:7, 11:7, 12:8, 13:9, 14:10, 15:11, 16:12, 17:13, 18:14, 19:15}


imglist = getFileList(image_dir, [])
print(len(imglist))
x = 0
for img_d in imglist:
    print(x)
    img_name = img_d.split('/')[-1]
    sample = img_d.split('/')[-2]
    image = cv2.imread(img_d, 0)

    image_t = transfer2(image, mapping)
    os.makedirs(os.path.join(transfer_image_written_dir, sample), exist_ok=True)

    cv2.imwrite(os.path.join(transfer_image_written_dir, sample, img_name), image_t)
    x+=1

`
The updated config files including ./configs/_base_/3D_texture_building.py, ./mmseg/datasets/__init__.py, ./mmseg/datasets/MIO.py, you can re-download these files to start training the building metaclass. Remember to update the mapped GT dir in data_prefix in the training config.

**Comment by PROPHETE-pro** [comment]
Thank you for pointing out the problem. Our dataset was initialled labelled with 20 classes and further refined to 16 (14 + bg + other) classes. It seems that we have uploaded the early version of the ground truth label and the config. Here I can provide you a mapping code that you can process the GT by yourself, and the correct config files.

The mapping code can be found at ./mmsegmentation/mapping.py

The updated config files including ./configs/_base_/3D_texture_building.py, ./mmseg/datasets/__init__.py, ./mmseg/datasets/MIO.py, you can re-download these files to start training the building metaclass. Remember to update the mapped GT dir in data_prefix in the training config.
I will re-upload the dataset with the correct label on the link very soon.

Sorry for the inconvenience anyway.

**Comment by zjtde1990** [comment]
Thank you for your patient answer. But I still have a question: When training for buildings, will the material categories not included in the building data affect the final training results? Is it necessary to exclude non-building material categories when calculating losses? Also, I found that the class-balanced sampling strategy mentioned in the paper is not used in the configuration file. Is it necessary for re-training to achieve good results?

**Comment by PROPHETE-pro** [comment]
Accroding to your first question, in our dataset, the existing categories are the same in training and testing, so the answer could be No. I suppose you are asking that if there are 10 categories in training and only 8 of them appears in testing GT, then the answer will be yes, as when testing there is a chance for the 2 missing categories to be miss-classified, and resulting in that category IOU = 0, which should be calculated in the mIoU. The reason there are NaN appears in the non-included categories is because both the GT and the inferenced results do not include that category. 

It is not necessary to exclude non-building categories, as the model actually does not know that there are non-building categories (as it is no where to be found in training data), the loss will not take those categories into consideration.

The class-balance sampling strategy we directly use the provided function in mmseg, you can found it in 
`custom3D_class = dict(
    type='ClassBalancedDataset',
    dataset=custom3D_train,
    oversample_thr=0.03
)`
you can replace the 'dataset' config in train_dataloader as 'custom3D_class' to enable sampling strategy. We conduct some ablations on this before, so it may not be changed back yet.
